fs/localfs/local_fs.go

```diff
@@ -254,6 +255,14 @@ func (fsf *filesystemFile) Open(ctx context.Context) (fs.Reader, error) {
 	return &fileWithMetadata{f}, nil
 }
 
+func (fsf *filesystemFile) DirEntryOrNil(ctx context.Context) (*snapshot.DirEntry, error) {
```

Comment:
it would be great to not implement this on all filesystemFiles to avoid the extra cost of Lstat() as we read the placeholder files (which most of the times won't be in use). The performance penalty is likely small (due to fs cache), but could be not insignificant at scale.

I think we could detect the special extension in `Readdir()` here and only return entries with `DirEntryOrNil` for files/subdirs that have `kopia-entry` extension.

WDYT?

Reply:
Fixed in 7f232537.